EventProc/src/HcalDigiProducer.cxx

```diff
@@ -128,76 +145,151 @@ namespace ldmx {
+
+            if (cur_subsection != 0){ // for sidecal don't worry about attenuation because it's single readout
+                hcalLayerPEs[detIDraw] = random_->Poisson(meanPE+meanNoise_);
+                hcalLayerMinPEs[detIDraw] = hcalLayerPEs[detIDraw];
+            }
+            if (cur_subsection == 0){// get PEs with attentuation
+                meanPE *= 100./80.; // increase the PE count to the case with no attentuation (assuming 80% attenuation on the pe_per_mip number)
+                float total_width = STRIPS_BACK_PER_LAYER_*50.0;
+                float distance_along_bar = 0;
+                if (cur_layer % 2 == 0){
+                    distance_along_bar = fabs(cur_ypos);
+                }
+                if (cur_layer % 2 == 1){
+                    distance_along_bar = fabs(cur_xpos);
+                }
+                float meanPE_close = meanPE * 0.8 * ((total_width/2. - distance_along_bar) / 1000.);
+                float meanPE_far   = meanPE * 0.8 * ((total_width/2. + distance_along_bar) / 1000.);
+                float PE_close     = random_->Poisson(meanPE_close+meanNoise_);
+                float PE_far       = random_->Poisson(meanPE_far+meanNoise_);
+                hcalLayerPEs[detIDraw] = PE_close + PE_far;
+                hcalLayerMinPEs[detIDraw] = std::min(PE_close,PE_far);
+            }
+            // std::cout << "depEnergy = " << depEnergy << "\t hcalLayerPEs[detIDraw] = " << hcalLayerPEs[detIDraw] << "\t hcalLayerMinPEs[detIDraw] = " << hcalLayerMinPEs[detIDraw] << std::endl;
+
+            if (cur_subsection == 0){
+                float super_strip_width = SUPER_STRIP_SIZE_*50.0;
+                float total_super_strips = (STRIPS_BACK_PER_LAYER_ + SUPER_STRIP_SIZE_ - 1)/SUPER_STRIP_SIZE_; // to help the integer round up
+                float total_width = STRIPS_BACK_PER_LAYER_*50.0;
+                if (cur_layer % 2 == 0){ // even layers, vertical
+                    cur_xpos = (super_strip_width * (float(cur_strip)+0.5)) - total_width/2.; 
+                    cur_ypos = hcalYpos[detIDraw] + random_->Gaus(0.,150.); 
+                }
+                if (cur_layer % 2 == 1){ // odd layers, horizontal
+                    cur_ypos = (super_strip_width * (float(cur_strip)+0.5)) - total_width/2.; 
+                    cur_xpos = hcalXpos[detIDraw] + random_->Gaus(0.,150.); 
+                }
+                if (cur_xpos > total_width/2.) cur_xpos = total_width/2.;
+                if (cur_xpos < -1.*total_width/2.) cur_xpos = -1.*total_width/2.;
+                if (cur_ypos > total_width/2.) cur_ypos = total_width/2.;
+                if (cur_ypos < -1.*total_width/2.) cur_ypos = -1.*total_width/2.;
+                // std::cout << "super_strip_width = " << super_strip_width << "\t total_super_strips = " << total_super_strips << "\t total_width = " << total_width << std::endl;
+            }
+
+            if( hcalLayerPEs[detIDraw] >= readoutThreshold_ ){ // > or >= ?
```

Comment:
>= is fine for now I think, this will depend on the readout scheme anyway.

Reply:
We had a detailed discussion about this in the software meeting maybe a week and a half ago.  The meaning of `readoutThreshold_` now strictly means, at what threshold do we save hits and that's it.  This is because before this parameter was dually used to also help generate noise.  This can get confusing for the 2 SiPM bar hit, so instead the noise generation for the HCal is fixed to generate noise above 1 PE.